transport-native-io_uring/src/main/java/io/netty/channel/uring/SockaddrIn.java

```diff
@@ -0,0 +1,165 @@
+     *     sa_family_t     sin6_family;   // AF_INET6
+     *     in_port_t       sin6_port;     // port number
+     *     uint32_t        sin6_flowinfo; // IPv6 flow information
+     *     struct in6_addr sin6_addr;     // IPv6 address
+     *     uint32_t        sin6_scope_id; /* Scope ID (new in 2.4)
+     * };
+     *
+     * struct in6_addr{
+     *     unsigned char s6_addr[16];   // IPv6 address
+     * };
+     */
+    static int writeIPv6(long memory, InetAddress address, int port) {
+        int written = 0;
+        // AF_INET6
+        PlatformDependent.putShort(memory, Native.AF_INET6);
+        written += 2;
+        PlatformDependent.putShort(memory + written, handleNetworkOrder((short) port));
+        written += 2;
+        PlatformDependent.putInt(memory + written, 0);
+        written += 4;
+        byte[] bytes = address.getAddress();
+        if  (bytes.length == 4) {
+            PlatformDependent.copyMemory(IPV4_MAPPED_IPV6_PREFIX, 0, memory + written, IPV4_MAPPED_IPV6_PREFIX.length);
+            written += IPV4_MAPPED_IPV6_PREFIX.length;
+            PlatformDependent.copyMemory(bytes, 0, memory + written, 4);
+            written += 4;
+            PlatformDependent.putInt(memory + written, 0);
+            written += 4;
+        } else {
+            PlatformDependent.copyMemory(bytes, 0, memory + written, 16);
+            written += 16;
+            PlatformDependent.putInt(memory + written, ((Inet6Address) address).getScopeId());
+            written += 4;
+        }
+        written += writePadding(memory + written, Native.SIZEOF_SOCKADDR_IN6 - written);
+        assert written == Native.SIZEOF_SOCKADDR_IN6;
+        return written;
+    }
+
+    static InetSocketAddress readIPv4(long memory, byte[] tmpArray) {
+        assert tmpArray.length == 4;
+        int port = handleNetworkOrder(PlatformDependent.getShort(memory + 2)) & 0xFFFF;
+        PlatformDependent.copyMemory(memory + 4, tmpArray, 0, 4);
+        try {
+            return new InetSocketAddress(InetAddress.getByAddress(tmpArray), port);
+        } catch (UnknownHostException ignore) {
+            return null;
+        }
+    }
+
+    static InetSocketAddress readIPv6(long memory, byte[] tmpArray) {
+        assert tmpArray.length == 16;
+        int port = handleNetworkOrder(PlatformDependent.getShort(memory + 2)) & 0xFFFF;
+        PlatformDependent.copyMemory(memory + 8, tmpArray, 0, 16);
+        int scopeId = PlatformDependent.getInt(memory + 24);
+        try {
+            return new InetSocketAddress(Inet6Address.getByAddress(null, tmpArray, scopeId), port);
+        } catch (UnknownHostException ignore) {
+            return null;
+        }
+    }
+
+    private static short handleNetworkOrder(short v) {
+        return BIG_ENDIAN_NATIVE_ORDER ? v : Short.reverseBytes(v);
+    }
+
+    /**
+     * Fill with {@code 0}s if any padding is needed.
+     */
+    private static int writePadding(long memoryAddress, int length) {
```

Comment:
any reason for not doing `return PlatformDependent.setMemory(memoryAddress, length, 0);`?

Reply:
no other reason then that I missed it :) Thanks for the tip :)